src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -197,6 +209,35 @@ std::optional<DataModel::ActionReturnStatus> CodegenDataModelProvider::InvokeCom
     return std::nullopt;
 }
 
+CHIP_ERROR CodegenDataModelProvider::GetNodeDataModelConfiguration(DataModel::NodeDataModelConfiguration & outConfig)
+{
+    uint32_t configurationVersion    = 0;
+    uint16_t size                    = sizeof(configurationVersion);
+    chip::StorageKeyName kStorageKey = chip::DefaultStorageKeyAllocator::ConfigurationVersion();
+    ReturnErrorOnFailure(mPersistentStorageDelegate->SyncGetKeyValue(kStorageKey.KeyName(), &configurationVersion, size));
+
+    outConfig.configurationVersion = configurationVersion;
+    outConfig.maxPathPerInvoke     = CHIP_CONFIG_MAX_PATHS_PER_INVOKE;
+    outConfig.dataModelVersion     = Revision::kDataModelRevision;
+    outConfig.specVersion          = Revision::kSpecificationVersion;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CodegenDataModelProvider::BumpNodeDataModelConfigurationVersion()
+{
+    uint32_t configurationVersion    = 0;
+    uint16_t size                    = sizeof(configurationVersion);
+    chip::StorageKeyName kStorageKey = chip::DefaultStorageKeyAllocator::ConfigurationVersion();
```

Comment:
Suggest not failing to record new one in memory even if the storage/access fails.

Reply:
Are you thinking of holding a copy in memoery, right now it expects to be able to read it from storage so if that fails here, how would it know what to bump and the read of the ConfigurationVersion attribute would also fail?

Reply:
But I'm still not sure how I feel about these getting out of sync, since while they are a reboot will revert the bump 🤔